pkg/utils/k8sutil.go

```diff
@@ -755,3 +757,79 @@ func addInitContainerAnnotation(dpm *v1beta1.Deployment) error {
 	}
 	return nil
 }
+
+// CreateIngress creates ingress rule for a specific function
+func CreateIngress(client kubernetes.Interface, ingressName, function, domain, ns string) error {
+	if domain == "" {
+		var err error
+		domain, err = getLocalDomain()
+		if err != nil {
+			return err
+		}
+	}
+
+	//TODO: skip annotation. We can add it later
+	//ingressAnnotations := map[string]string{
+	//	"kubernetes.io/ingress.class": "nginx",
+	//	"kubernetes.io/tls-acme":      "true",
+	//}
+
+	ingress := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ingressName,
+			Namespace: ns,
+			//Annotations: ingressAnnotations,
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: fmt.Sprintf("%s.%s", ingressName, domain),
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Path: "/",
+									Backend: v1beta1.IngressBackend{
+										ServiceName: function,
+										ServicePort: intstr.FromInt(8080),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	_, err := client.ExtensionsV1beta1().Ingresses(ns).Create(ingress)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// getLocalDomain returns hostname
+func getLocalDomain() (string, error) {
```

Comment:
erm. what?
This function gets the kubernetes API endpoint, assumes it's an IP-literal and then appends ".nip.io".  That's only going to work in a very specific set of circumstances, particularly if you then want to do letsencrypt TLS against this domain name (which is never going to work with nip.io).
I'd have to think about what makes a good robust default, but I feel like this solution is worse than just forcing the user to tell us a valid hostname.